1주차/4.가계부/script.js

```diff
@@ -1,25 +1,379 @@
+}
+
+function filter_item(show_income, show_outcome) {
+  return HISTORY_LIST.filter((item) => {
+    if (show_income && show_outcome) {
+      return true;
+    }
+    if (show_income) return item.price > 0;
+    if (show_outcome) return item.price < 0;
+    return false;
+  });
+}
+
+const cancel_modal = document.querySelector(".cancel_modal");
+
+incomeButton.addEventListener("change", filter_history);
+outcomeButton.addEventListener("change", filter_history);
+
+let total_income = 0;
+let total_outcome = 0;
+let total_money = null;
+const total = document.querySelector(".total");
+const income = document.querySelector(".income");
+const outcome = document.querySelector(".outcome");
+function show_list(history_list) {
+  initial_list.innerHTML = ""; // 이전 내역 지우기
+  total_money = null;
+  total_income = 0;
+  total_outcome = 0;
+  const list_real = document.querySelector(".list-real");
+
+  history_list.forEach((item) => {
+    const cancelButton = document.createElement("button");
+    cancelButton.classList.add("x");
+    cancelButton.textContent = "x";
+    cancelButton.addEventListener("click", function () {
+      cancel_modal.style.display = "block";
+      const listItem = this.parentNode;
+      YesCancel.addEventListener("click", function handler() {
+        console.log(listItem);
+        const removeItem = history_list.find((x) => x.content == item.content);
+        console.log(removeItem);
+        if (removeItem.price > 0) {
+          total_income -= removeItem.price;
+        }
+        if (removeItem.price < 0) {
+          total_outcome -= removeItem.price;
+        }
+
+        total_money = total_income - total_outcome;
+        total.textContent = `${total_money.toLocaleString()}원`;
+        income.textContent = `+${total_income.toLocaleString()}`;
+        outcome.textContent = `${total_outcome.toLocaleString()}`;
+        listItem.remove();
+        cancel_modal.style.display = "none";
+        // YesCancel.removeEventListener("click", handler);
+      });
+      NoCancel.addEventListener("click", function () {
+        cancel_modal.style.display = "none";
+      });
+    });
+    const ul = document.createElement("ul");
+    ul.classList.add("list-indiv");
+
+    const category_li = document.createElement("li");
+    category_li.classList.add("category");
+    category_li.textContent = item.category;
+
+    const content_li = document.createElement("li");
+    content_li.classList.add("content");
+    content_li.textContent = item.content;
+
+    const price_li = document.createElement("li");
+    price_li.classList.add("price");
+    price_li.textContent =
+      item.price < 0
+        ? `${item.price.toLocaleString()}원`
+        : `+${item.price.toLocaleString()}원`;
+    if (item.price < 0) {
+      price_li.style.color = "blue";
+      total_outcome += item.price;
+      // console.log(item.price, "가격");
+    } else {
+      price_li.style.color = "red";
+      total_income += item.price;
+    }
```

Comment:
`item.price`에 관한 조건 처리가 조금 겹치는 부분이 있는 것 같아서 ..! 이런 식으로 수정해봤어!

```suggestion
    if (item.price < 0) {
      price_li.style.color = "blue";
      price_li.textContent = `${item.price.toLocaleString()}원`
      total_outcome += item.price;
    } else {
      price_li.style.color = "red";
      price_li.textContent = `+${item.price.toLocaleString()}원`;
      total_income += item.price;
    }
```

나는 중복 코드를 최대한 없애는걸 좋아하는 편이라 이렇게 수정해봤는데,
개인적인 의견이라 참고정도로 생각해도 좋을 것 같아 !